src/test/func/malloc/malloc.cc

```diff
@@ -0,0 +1,127 @@
+#include <cassert>
+#include <cerrno>
+#include <ctime>
+#include <malloc.h>
+#include <snmalloc.h>
+#include <test/xoroshiro.h>
+
+using namespace snmalloc;
+
+constexpr size_t min_pow_2 = bits::ctz_const(sizeof(size_t));
+constexpr size_t max_pow_2 = bits::ctz_const(SUPERSLAB_SIZE);
+
+size_t rand_pow_2(xoroshiro::p128r64& r, size_t min_pow = min_pow_2)
+{
+  const size_t max_pow = max_pow_2 - min_pow;
+  auto p = (r.next() % (max_pow + 1)) + min_pow;
+  return (size_t)1 << p;
+}
+
+size_t rand_non_pow_2(xoroshiro::p128r64& r, size_t min = 0)
+{
+  const size_t max = (1 << max_pow_2) - min;
+  auto n = (r.next() % (max + 1)) + min;
+  while (__builtin_popcount(n) < 2)
+    n = r.next();
+
+  return n;
+}
+
+void check_result(size_t size, void* p, int err, bool null)
+{
+  fprintf(stderr, "%p, errno: %s\n", p, strerror(errno)); // TODO
+  assert(errno == err);
```

Comment:
As you are using assert, this will not be tested in a release build.  Is that what you want?  Should we make this if (!b) abort().  So we can test the release build too?

Reply:
Or redefine assert